internal/resources/google/sql_database_instance.go

```diff
@@ -0,0 +1,316 @@
+package google
+
+import (
+	"github.com/infracost/infracost/internal/resources"
+	"github.com/infracost/infracost/internal/schema"
+
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/shopspring/decimal"
+)
+
+type SQLDatabaseInstance struct {
+	Address              string
+	DiskSize             int64
+	UseIPV4              bool
+	ReplicaConfiguration string
+	Tier                 string
+	AvailabilityType     string
+	Region               string
+	DatabaseVersion      string
+	DiskType             string
+	BackupStorageGB      *float64 `infracost_usage:"backup_storage_gb"`
+}
+
+var SQLDatabaseInstanceUsageSchema = []*schema.UsageItem{{Key: "backup_storage_gb", ValueType: schema.Float64, DefaultValue: 0}}
+
+func (r *SQLDatabaseInstance) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
+}
+
+func (r *SQLDatabaseInstance) BuildResource() *schema.Resource {
+	var resource *schema.Resource
+
+	replica := false
+	if r.ReplicaConfiguration != "" {
+		replica = true
+	}
+
+	resource = r.sqlDatabaseInstanceCostComponents(false, r.Address)
+	if replica {
+		resource.SubResources = append(resource.SubResources, r.sqlDatabaseInstanceCostComponents(true, "Replica"))
+	}
+
+	return resource
+}
+
+type SQLInstanceDBType int
+
+const (
+	MySQL SQLInstanceDBType = iota
+	PostgreSQL
+	SQLServer
+)
+
+func (r *SQLDatabaseInstance) sqlDatabaseInstanceCostComponents(replica bool, name string) *schema.Resource {
+	var costComponents []*schema.CostComponent
+	tier := r.Tier
+
+	availabilityType := "ZONAL"
+	if r.AvailabilityType != "" && !replica {
+		availabilityType = r.AvailabilityType
+	}
```

Comment:
```suggestion
	if r.AvailabilityType == "" || replica {
		r.AvailabilityType = "ZONAL"
	}
```
Should we change this so it sets the default struct property? That way we don't need to pass it to the the methods like `r.memoryCostComponent` and `r.cpuCostComponent` and we can just access it directly from them?

Reply:
Nice idea! I change this and few other fields like this.